source/adapter/alter.js

```diff
@@ -1,22 +1,41 @@
 var fetch = require("node-fetch");
+var deepmerge = require("deepmerge");
 
 var responseHandlers = require("./response.js");
 
+function getAlterItemDefaults() {
+    return {
+        headers: { },
+    };
+}
+
 module.exports = {
 
-    deleteItem: function deleteItem(url, targetPath) {
+    deleteItem: function deleteItem(url, targetPath, options) {
+        options = deepmerge.all([
+            getAlterItemDefaults(),
+            options || {}
+        ]);
         return fetch(url + targetPath, {
-                method: "DELETE"
+                method: "DELETE",
+                headers: options.headers
             })
             .then(responseHandlers.handleResponseCode);
     },
 
-    moveItem: function moveItem(url, filePath, targetFilePath) {
-        return fetch(url + filePath, {
-                method: "MOVE",
+    moveItem: function moveItem(url, filePath, targetFilePath, options) {
+        options = deepmerge.all([
+            getAlterItemDefaults(),
```

Comment:
Same here, might be the wrong title. Having separate functions for each is best as they may differ in options eventually.